src/battle_script_commands.c

```diff
@@ -16574,3 +16607,46 @@ void BS_SetPhotonGeyserCategory(void)
     gBattleStruct->swapDamageCategory = (GetCategoryBasedOnStats(gBattlerAttacker) == DAMAGE_CATEGORY_PHYSICAL);
     gBattlescriptCurrInstr = cmd->nextInstr;
 }
+
+void BS_TryUpdateRecoilTracker(void)
+{
+    NATIVE_ARGS();
+    u32 i;
+
+    if (GetBattlerSide(gBattlerAttacker) == B_SIDE_PLAYER
+     && !(gBattleTypeFlags & (BATTLE_TYPE_LINK
+                             | BATTLE_TYPE_EREADER_TRAINER
+                             | BATTLE_TYPE_RECORDED_LINK
+                             | BATTLE_TYPE_TRAINER_HILL
+                             | BATTLE_TYPE_FRONTIER)))
+    {
+        const struct Evolution *evolutions = GetSpeciesEvolutions(gBattleMons[gBattlerAttacker].species);
+        if (evolutions == NULL)
+        {
+            gBattlescriptCurrInstr = cmd->nextInstr;
+            return;
+        }
+        for (i = 0; evolutions[i].method != EVOLUTIONS_END; i++)
+        {
+            if (SanitizeSpeciesId(evolutions[i].targetSpecies) == SPECIES_NONE)
+                continue;
+            if (evolutions[i].method == EVO_LEVEL_RECOIL_DAMAGE)
+            {
+                u16 val;
+                val = GetMonData(&gPlayerParty[gBattlerPartyIndexes[gBattlerAttacker]], MON_DATA_EVOLUTION_TRACKER);
+                if (val == 512) // to account for the fact we only have 10 bits
+                    break;
+                if ((val + gBattleMoveDamage) > 512)
+                    val = 512;
+                else
+                    val += gBattleMoveDamage;
```

Comment:
Oh, shouldn't this be `511`? Also, I think (but please double-check!) you could write it like:
```suggestion
                val = min(511, val + gBattleMoveDamage);
```